### ISSUE TITLE
Kong Error Invalid authentication credentials

### ISSUE THREAD

**Issue by tagNC** [supabase/supabase#19713]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I have followed all setup [(here)](https://supabase.com/docs/guides/self-hosting/docker#before-you-begin) and I keep having the follwing error after login : 
`Kong Error Invalid authentication credentials.`

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to my proxy URL 'https://supabase.<mydomain>.com/' (in .env is the value of SITE_URL)
2. Enter credentials (DASHBAOARD_USERNAME and DASHBOARD_PASSWORD)
3. See error `Kong Error Invalid authentication credentials.`

## Expected behavior

Access to the Dashboard.

## Screenshots

If applicable, add screens
![supabase1](https://github.com/supabase/supabase/assets/150861045/90c3aa8f-2259-44d6-83a6-e1ce2e4e0869)
![supabase2](https://github.com/supabase/supabase/assets/150861045/67c3609c-dd98-457b-9509-96c0b3533c24)
hots to help explain your problem.

## System information

- OS: 
PRETTY_NAME="Ubuntu 23.10"
NAME="Ubuntu"
VERSION_ID="23.10"
VERSION="23.10 (Mantic Minotaur)"
VERSION_CODENAME=mantic
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=mantic
LOGO=ubuntu-logo

- Browser Chrome, Chrome private, Firefox, Firefox private
- Version of supabase-js: latest

## Additional context

Here is the `docker compose ps` result : 
`NAME                             IMAGE                              COMMAND                                                                                                                                                 SERVICE     CREATED       STATUS                      PORTS
realtime-dev.supabase-realtime   supabase/realtime:v2.25.35         "/usr/bin/tini -s -g -- /app/limits.sh sh -c '/app/bin/migrate && /app/bin/realtime eval 'Realtime.Release.seeds(Realtime.Repo)' && /app/bin/server'"   realtime    4 hours ago   Up 17 minutes (healthy)
supabase-analytics               supabase/logflare:1.4.0            "sh run.sh"                                                                                                                                             analytics   4 hours ago   Up 17 minutes (healthy)     0.0.0.0:4000->4000/tcp, :::4000->4000/tcp
supabase-auth                    supabase/gotrue:v2.125.1           "gotrue"                                                                                                                                                auth        4 hours ago   Up 17 minutes (healthy)
supabase-db                      supabase/postgres:15.1.0.117       "docker-entrypoint.sh postgres -c config_file=/etc/postgresql/postgresql.conf -c log_min_messages=fatal"                                                db          4 hours ago   Up 17 minutes (healthy)     0.0.0.0:5432->5432/tcp, :::5432->5432/tcp
supabase-edge-functions          supabase/edge-runtime:v1.29.1      "edge-runtime start --main-service /home/deno/functions/main"                                                                                           functions   4 hours ago   Up 17 minutes
supabase-kong                    kong:2.8.1                         "bash -c 'eval \"echo \\\"$(cat ~/temp.yml)\\\"\" > ~/kong.yml && /docker-entrypoint.sh kong docker-start'"                                             kong        4 hours ago   Up 17 minutes (healthy)     0.0.0.0:8000->8000/tcp, :::8000->8000/tcp, 8001/tcp, 0.0.0.0:8443->8443/tcp, :::8443->8443/tcp, 8444/tcp
supabase-meta                    supabase/postgres-meta:v0.75.0     "docker-entrypoint.sh npm run start"                                                                                                                    meta        4 hours ago   Up 17 minutes (healthy)     8080/tcp
supabase-rest                    postgrest/postgrest:v11.2.2        "postgrest"                                                                                                                                             rest        4 hours ago   Up 17 minutes               3000/tcp
supabase-storage                 supabase/storage-api:v0.43.11      "docker-entrypoint.sh node dist/server.js"                                                                                                              storage     4 hours ago   Up 17 minutes (healthy)     5000/tcp
supabase-studio                  supabase/studio:20231123-64a766a   "docker-entrypoint.sh node apps/studio/server.js"                                                                                                       studio      4 hours ago   Up 17 minutes (unhealthy)   3000/tcp
supabase-vector                  timberio/vector:0.28.1-alpine      "/usr/local/bin/vector --config etc/vector/vector.yml"                                                                                                  vector      4 hours ago   Up 17 minutes (healthy)
`

**Comment by tagNC** [comment]
I could notice that as soon as I try to update DASHBOARD_USERNAME and DASHBOARD_PASSWORD in .env file, the error is happening.

**Comment by tagNC** [comment]
Changing credentials in the kong.yml and let credentials blank in .env is make it work.

**Comment by zj1123581321** [comment]
> Changing credentials in the kong.yml and let credentials blank in .env is make it work.

You mean:
` volumes/api/kong.yml ` should change to this?

```
consumers:
  - username: DASHBOARD
  - username: anon
    keyauth_credentials:
      - key: <your_generated_anon_key>
  - username: service_role
    keyauth_credentials:
      - key: <your_generated_service_role_key>
```

and let `.env` change to this:
```
############
# Secrets
# YOU MUST CHANGE THESE BEFORE GOING INTO PRODUCTION
############

POSTGRES_PASSWORD=your-super-secret-and-long-postgres-password
JWT_SECRET=your-super-secret-jwt-token-with-at-least-32-characters-long
ANON_KEY=
SERVICE_ROLE_KEY=
DASHBOARD_USERNAME=supabase
DASHBOARD_PASSWORD=this_password_is_insecure_and_should_be_updated
```

I'm not certain if I've grasped it correctly. I'm facing the same problem and haven't resolved it yet.

looking forward to your response and greatly appreciate it!

**Comment by Stokestack** [comment]
The problem I have is that there's no way to clear this error. I actually did enter wrong credentials, but the page offers no way to try again, and clearing browser data associated with localhost doesn't clear it.

So now there's no way to access the dashboard.

**Comment by Glinte** [comment]
Seems like you can get around it by putting the login credentials in the url: https://stackoverflow.com/questions/50528467/how-to-add-login-credentials-to-url
i.e. http://username:password@127.0.0.1:8000/

**Comment by Stokestack** [comment]
Thanks for the temporary workaround.

Why was this closed as completed? What was changed?

**Comment by Khaled-Mo99** [comment]
I'm still facing the same issue and event the workaround not work for me

**Comment by Stokestack** [comment]
Other people [continue to encounter this problem](https://github.com/orgs/supabase/discussions/27895)

**Comment by qxygene1** [comment]
just change password and leave username as supabase and it worked

**Comment by Stokestack** [comment]
You can't, because the log-in fields aren't presented anymore.

**Comment by qxygene1** [comment]
> You can't, because the log-in fields aren't presented anymore.

worked for me

**Comment by Stokestack** [comment]
Well, not for others. Hence this bug report.

**Comment by freedomanonyme** [comment]
Well This is a false start with me with Supabase. Don't have the time two stick with this king of issue.

**Comment by c-o-l-i-n** [comment]
This workaround worked for me:

1. Open the Supabase dashboard **with the username and password in the URL**.
This takes you to the dashboard, but none of the data loads properly.
_eg: https://username:password@your-supabase-dashboard-domain.com_

3. Close that tab, and then open a _new_ tab **without** the username and password.
Now the data should load!
_eg: https://your-supabase-dashboard-domain.com_

**Comment by djbouti** [comment]
Ran into this github pages, I have to tell you guys to chill. It is simple authentication, like, really simple that it read the cookies first. If your cookies are 'invalid password' cookies, then it will always show invalid authentication. WHY? Because it read the cookies before taking in your input (password).

Clear the cookies for the supabase (usually localhost:8000), close the browser, reopen the browser and go back to your supabase (usually localhost:8000) and this time make sure to input it correctly.

You are welcome.

**Comment by bastien-naturavelo** [comment]
Well, it's pretty random, on a computer i'm logged in with one browser, on a different browser it never showed me the popup to login and there's no cookie attached, it doesn't work either on private browsing. 
And on another computer, i get the login but the credentials are not working anymore. It's disturbing.

**Comment by amgawishx** [comment]
This issue still occurs, why was it closed? what don't they fix it? why the work arounds? there is obviously something wrong here.

**Comment by jljl1337** [comment]
It might not be directly relevant, but does it make more sense to expose the dashboard on a different port? Anyone with the public url can try to login to the dashboard.

**Comment by bikos** [comment]
**THIS IS FOR SUPABASE RUNNING  THROUGH COOLIFY BUT SHOULD BE SAME FOR OTHER USERS TOO:**

This might help other poor souls who ran through this again. 

* How to  First find the YML file.

  Find the container ID running kong. 
  `docker ps`

   Then

   `docker inspect container_name_or_id`

* Find the path for kong. yml file

  ![Image](https://github.com/user-attachments/assets/f27a5597-b0be-427a-a082-5fc195545528)


* View the YML file


![Image](https://github.com/user-attachments/assets/230d5fe9-4e2c-4373-9c94-223f7906fad4)

 Either hard cord your username/password or update in your .env. Coolify gives a nice UI to update your environment variable. 

I did it as follows and it worked like a charm:

<img width="961" alt="Image" src="https://github.com/user-attachments/assets/37935ddb-9599-4f85-b1cb-b08907f016a4" />

**Comment by shizuo-x** [comment]
> This workaround worked for me:
> 
>     1. Open the Supabase dashboard **with the username and password in the URL**.
>        This takes you to the dashboard, but none of the data loads properly.
>        _eg: https://username:password@your-supabase-dashboard-domain.com_
> 
>     2. Close that tab, and then open a _new_ tab **without** the username and password.
>        Now the data should load!
>        _eg: https://your-supabase-dashboard-domain.com_

works, I swear man such a bs issue, I respect everyone but imagine a noob like me trying to do something in life only to get to this seemingly, easily, fixable problem; but waste hours to troubleshoot.

**Comment by lucca-rodrigues** [comment]
Apparently this happens if you are already logged in with the old credentials (before changing)

I had the same problem, and when accessing in an incognito tab I was able to log in normally with the new credentials

**Comment by infiniteAppsUG** [comment]
> Apparently this happens if you are already logged in with the old credentials (before changing)
> 
> I had the same problem, and when accessing in an incognito tab I was able to log in normally with the new credentials

THIS IS THE ISSUE. LAYER 8 :D thanks for the hint. Worked out of the box

**Comment by JamesMowery** [comment]
This is still a mess. I have cleared cookies, restarted browser, and I can't get it to allow me on the login page, just because I screwed up typing in the credentials one single time. Please fix this!

Or, better yet, why not have an actual normal login screen?

**Comment by neurabluAI** [comment]
> This is still a mess. I have cleared cookies, restarted browser, and I can't get it to allow me on the login page, just because I screwed up typing in the credentials one single time. Please fix this!
> 
> Or, better yet, why not have an actual normal login screen?

Have you tried a private browser Session? This worked for me at least.

**Comment by sIRwa2** [comment]
pretty wild this is marked closed with the fix as suggested.. For me, seeing the error in firefox: > clear cache and cookies for http://192.. solved it. indeed not a pleasant introduction to supabase.

**Comment by manuelebeh** [comment]
I don't know if this will help, but here's what worked for me.

**Problem**

After deleting and completely recreating my Supabase instance on Dokploy, I noticed that access to my Supabase host (supabase.domain.com was no longer working properly. Every attempt to connect resulted in an error such as: `Kong Error: Invalid authentication credentials `

In practice, this meant that Kong was still receiving old credentials or session tokens stored in my browser before the instance was deleted. Even though the new Supabase instance was correctly configured, the browser kept sending outdated authentication data, which prevented any successful connection.

**Analysis**

- Supabase (and Kong) manages authentication via cookies and browser storage (localStorage, sessionStorage).
- After deleting/recreating the instance, all previous JWT keys and sessions become invalid.
- However, the browser continued to reuse these old values, which caused the Invalid authentication credentials error.

I tried accessing the site in private/incognito mode and logging in again with the correct credentials, but the issue still persisted in my normal browser session.

I tried clearing the cache and cookies in a generic way, but that alone did not resolve the issue.

**Solution**

What finally worked was fully deleting all browser data for the affected domain:

1. Open your browser Settings
2. Go to Privacy / Cookies and site data
3. Search for your Supabase domain or main domain
4. Delete all stored data for that domain
5. Reload the page

After doing this, the browser no longer contained any old tokens or sessions, and access worked normally using the new keys and tokens from the recreated Supabase instance.
